### DISCUSSION TITLE
Does this have context of other open files or only the one that is open?

### DISCUSSION THREAD

**Discussion by prabhuvi** [intitni/CopilotForXcode#466] [General]
Looking for an answer to couple of questions.
- Does the extension have access to all open files in Xcode and not just the one that is currently active?
- Is there anything done explicitly to pass the conversation history and not just the current question with the context from the chat window?
- Does the extension have access to the entire workspace (all files in the workspace), trying to know about file indexing.

Thank you in advance!

**Comment by intitni** [comment]
For the suggestion feature:

Both GitHub Copilot and Codeium know about the opened files in the same workspace and may use them to generate suggestions.

For the chat feature:

The conversation history is included by default. You can control the number of messages to include from the host app. The app will truncate the history when it hits the context window of the LLM.

By default (`code` scope), the app will include the focused code (usually the function or type definition covering the text cursor) in the conversation. 

If the `sense` scope is on, the app will include the interfaces of the types referenced by the focused code in the conversation.

If the `project` scope is on, the app will include relevant files that contain the keywords from your questions in the conversation. But there is no pre-indexing at the moment, the app has to search for the files when requested.

**Reply by prabhuvi** [reply]
Is there a scope that captures the other open files in the context?

**Reply by intitni** [reply]
There is no such scope at the moment. What are you trying to accomplish?

There is a planned feature to include a specific piece of code by mentioning the symbol name or file name in the question.
